code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -336,7 +336,7 @@
 
 
 /mob/living/simple_animal/hostile/proc/Shoot(atom/targeted_atom)
-	if(targeted_atom == targets_from.loc || targeted_atom == targets_from)
+	if(targeted_atom == targets_from.loc || targeted_atom == targets_from || targeted_atom == null)
```

Comment:
Use `!targeted_atom` as the first check. No point in trying to compare something that does not exist with other things before checking if that thing even exists!